core/src/main/scala/com/stripe/agate/tensor/Storage.scala

```diff
@@ -212,49 +291,57 @@ object Storage {
       maxChunkSizeInBytes: Int
   ): Resource[IO, Storage[dt.Elem]] = {
     val tb = ToBytes.forDataType(dt)
-    val step = tb.size
-    if (maxChunkSizeInBytes < step) {
-      Resource.liftF(
-        IO.raiseError(
-          new IllegalArgumentException(s"maxChunkSizeInBytes=${maxChunkSizeInBytes} < $step")
+    tb.strategy match {
+      case _: ToBytes.Strategy.VarLength[_] =>
+        Resource.liftF(
+          IO.raiseError(
+            new IllegalArgumentException(s"unsupported datatype: $dt")
```

Comment:
Currently just chucking an exception. We'll need a different strategy for variable-length encoded values instead of using a `ChunkedStorage`. I'm going to do this in a separate PR though. We already throw for anything except `Float32` anyways, so I figure it's OK to throw for strings for the near term too.